docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
 # Developer Guide
 
-## Design & Implementation
+### Table of Contents
+* **[1. Introduction](#1-introduction)**
+* **[2. Overview of the SHOCO application](#2-overview-of-the-shoco-application)**
+* **[3. Implementation](#3-implementation)**
+    + [3.1 Add feature](#31-add-feature)
+    + [3.2 Edit feature](#32-edit-feature)
+    + [3.3 Delete feature](#33-delete-feature)
+    + [3.4 Find feature](#34-find-feature)
+    + [3.5 Mark and Unmark feature](#35-mark-and-unmark-feature)
+    + [3.6 Display feature](#36-display-feature)
+    + [3.7 Clear list feature](#37-clear-list-feature)
+    + [3.8 Set budget feature](#38-set-budget-feature)
+    + [3.9 Reset budget feature](#39-reset-budget-feature)
+    + [3.10 View help feature](#310-view-help-feature)
+    + [3.11 Exit program feature](#311-exit-program-feature)
+ * **[Appendix A: Product Scope](#appendix-a-product-scope)**
+ * **[Appendix B: User Stories](#appendix-b-user-stories)**
+ * **[Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)**
+ * **[Appendix D: Glossary](#appendix-d-glossary)**
+ * **[Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)**
+ 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
 
+### Purpose of this guide
+This guide describes the software architecture and design of the SHOCO application.
+It will evolve throughout the design and implementation of each SHOCO release. 
+Currently, this document is for the first public release of the application, SHOCO v1.0.
+
+### Scope of this guide
+This document describes the software architecture and design for the implementation
+of SHOCO and is tailored for the developers, designers, and software testers of SHOCO.
+
+&nbsp;
+
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+&nbsp;
+
+## 2. Overview of the SHOCO application
+
+The overview of the main classes in the application are shown in the class diagram below.
+Omitted are the classes for the features implemented, the <code>LoadData</code>
+class, <code>WriteData</code> class, <code>FileUtil</code> class and
+<code>CommandLineTable</code> class.
+
+![alt text](images/ClassDiag.png)
```

Comment:
Maybe here lacks of the some Associations. e.g : In your Parser, there is a private newCommand, so there should be one link between Parser and Command.
<img width="888" alt="image" src="https://user-images.githubusercontent.com/38337361/77996957-323b7200-7361-11ea-933d-acca6cfd6a9a.png">